datanode/api/trading_data_v2.go

```diff
@@ -103,22 +102,22 @@ func (t *tradingDataServiceV2) ListAccounts(ctx context.Context, req *v2.ListAcc
 	defer metrics.StartAPIRequestAndTimeGRPC("ListAccountsV2")()
 	pagination, err := entities.CursorPaginationFromProto(req.Pagination)
 	if err != nil {
-		return nil, apiError(codes.InvalidArgument, ErrMalformedRequest, err)
+		return nil, formatE(err, ErrInvalidPagination.Error())
```

Comment:
Before this was returning `codes.InvalidArgument`, but now it'll be `code.InternalError` when the error could be something like `pagination limit must be greater than 0`.